gadget/validate.go

```diff
@@ -216,6 +250,14 @@ func validateVolumeContentsPresence(gadgetSnapRootDir string, vol *LaidOutVolume
 			continue
 		}
 		for _, c := range s.Content {
+			if strings.HasPrefix(c.UnresolvedSource, "$kernel:") {
+				// This only validates that the ref is valid.
+				// Resolving happens with ResolveContentPaths()
+				if _, _, err := splitKernelRef(c.UnresolvedSource); err != nil {
+					return fmt.Errorf("cannot use kernel reference %q: %v", c.UnresolvedSource, err)
+				}
+				continue
+			}
 			// TODO: detect and skip Content with "$kernel:" style refs if there is no kernelSnapRootDir passed in as well
```

Comment:
feels like this TODO will still need addressing later, and so it needs to be moved before the new if

Reply:
done